DEVELOPERS.md

```diff
@@ -0,0 +1,439 @@
+[^]  a6mpuk7          ^hdf5@1.14.1-2%intel@19.0.5.281~cxx~fortran+hl~ipo~java~map+mpi+shared~szip~threadsafe+tools api=default build_system=cmake build_type=Release generator=make arch=linux-rocky8-x86_64
+[^]  arj4bfz              ^zlib@1.2.13%intel@19.0.5.281+optimize+pic+shared build_system=makefile arch=linux-rocky8-x86_64
+[^]  mnx4ggh      ^netcdf-fortran@4.5.2%intel@19.0.5.281~doc+pic+shared build_system=autotools patches=b050dbd arch=linux-rocky8-x86_64
+[^]  sh2anmt      ^oasis3-mct@git.2023.11.09=2023.11.09%intel@19.0.5.281~deterministic~optimisation_report build_system=makefile arch=linux-rocky8-x86_64
+[e]  4jwtg3n      ^openmpi@4.0.2%intel@19.0.5.281~atomics~cuda~cxx~cxx_exceptions~gpfs~internal-hwloc~java~legacylaunchers~lustre~memchecker~orterunprefix+romio+rsh~singularity+static+vt+wrapper-rpath build_system=autotools fabrics=none patches=073477a,60ce20b schedulers=none arch=linux-rocky8-x86_64
+[^]  nwolfzy      ^parallelio@2.5.2%intel@19.0.5.281+fortran~ipo~logging+mpi~pnetcdf~shared~timing build_system=cmake build_type=Release generator=make patches=55a6d7a arch=linux-rocky8-x86_64
+
+==> Updating view at /g/data/tm70/aph502/dev_instructions/envs/mom5-cice5-dev/.spack-env/view
+==> Warning: Skipping external package: openmpi@4.0.2%intel@19.0.5.281~atomics~cuda~cxx~cxx_exceptions~gpfs~internal-hwloc~java~legacylaunchers~lustre~memchecker~orterunprefix+romio+rsh~singularity+static+vt+wrapper-rpath build_system=autotools fabrics=none patches=073477a,60ce20b schedulers=none arch=linux-rocky8-x86_64/4jwtg3n
+```
+
+Now there are source subdirectories for `mom5` and `cice5` 
+
+```bash
+$ ls -lg
+total 28
+drwxr-s--- 20 tm70  4096 Apr 30 20:49 cice5
+drwxr-s--- 11 tm70  4096 Apr 30 20:49 mom5
+-rw-r-----  1 tm70 15477 Apr 30 20:50 spack.lock
+-rw-r-----  1 tm70  1586 Apr 30 20:50 spack.yaml
+```
+
+Both can be compiled with a single call to `spack install`.
+
+### Modifying an external dependency
+
+External dependencies are packages that the model depends on but are not model components that ACCESS-NRI is responsible for. What is an external dependency and what is not can change. If there is a bug in an external dependency then it might be necessary to use `spack develop` to modify, build and test that dependency.
+
+Use `spack info` to determine what versions are supported for the `spack` version being used, and choose appropriate values.
+
+For illustrative purposes the latest versions of `netcdf-c` and `netcdf-fortran` are installed.
+
+```bash
+$ spack install netcdf-c@4.9.2 netcdf-fortran@4.6.0
+```
+
+This entailed compiling a number of dependent packages.
+
+The required versions of `netcdf-c` and `netcdf-fortran` were then updated in `spack.yaml`
+
+UNRESOLVED: Can't get this to work. Can compile the new versions outside an environment, but when I change the versions in the `spack.yaml` get
```

Comment:
Will this be resolved before merge?

Reply:
Yes, or removed. But it is a very obvious use case.

Reply:
Added merge warning